src/Vendr.DemoStore/Web/Controllers/SearchSurfaceController.cs

```diff
@@ -81,29 +82,38 @@ public ActionResult Search(string q = "", int p = 1, int ps = 12)
                     }
                 }
 
-                // Perform a faceted search based on search categories
-                using (var reader = IndexReader.Open(((LuceneIndex)index).GetLuceneDirectory(), true))
-                using (var factedSearcher = new SimpleFacetedSearch(reader, new string[] { "searchCategory" }))
-                {
-                    var queryParser = new QueryParser(Version.LUCENE_30, "", new KeywordAnalyzer());
-                    var query = queryParser.Parse(sb.ToString());
-                    var queryResults = factedSearcher.Search(query, ps * p);
+                query.And().ManagedQuery(sb.ToString());
+            }
+            
+            // Search for the category alias
+            if (request.Category.IsNullOrWhiteSpace() == false)
+            {
+                query.And().Field("__Search_categories", request.Category);
+            }
 
-                    var facetedResults = new Dictionary<string, PagedResult<IPublishedContent>>();
+            // Fetch facets for categories
+            query.And().Facet("categories");
 
-                    foreach (SimpleFacetedSearch.HitsPerFacet hpg in queryResults.HitsPerFacet)
-                    {
-                        facetedResults.Add(hpg.Name.ToString(), new PagedResult<IPublishedContent>(hpg.HitCount, p, ps)
-                        {
-                            Items = hpg.Documents.Skip(ps * (p - 1)).Select(x => UmbracoContext.Content.GetById(int.Parse(x.Get("id")))).ToList()
-                        });
-                    }
+            // Execute search
+            var searchResults = query.Execute();
 
-                    result.Facets = facetedResults;
-                }
-            }
+            var results = searchResults
+                .GetResults<IPublishedContent>()
+                .Skip(request.PerPage * (request.Page - 1))
+                .Take(request.PerPage);
+
+            model.Results = new PagedResult<IPublishedContent>(searchResults.TotalItemCount, request.Page, request.PerPage)
+            {
+                Items = results
+            };
+
+            var categories = searchResults
+                .GetFacet("categories")
+                .ToDictionary(x => UmbracoContext.Content.GetById(Udi.Parse(x.Value.ToString())), x => x.Hits);
```

Comment:
Could this be simplified?

```
var totalResults = searchResults.TotalItemCount;
var pagedResults = searchResults.Skip(request.PerPage * (request.Page - 1));

var items = pagedResults.ToPublishedSearchResults(_umbracoContextAccessor.UmbracoContext.Content)
                                        .Select(x => x.Content);

model.Results = new PagedResult<IPublishedContent>(searchResults.TotalItemCount, request.Page, request.PerPage)
{
        Items = items
};
```

Then `items` would be a collection `IEnumerable<IPublishedContent>>` and it might not be necessary with `UmbracoContext.Content.GetById` in the following lines, but just select to `IPublishedContent` entry from the collection.

```
var categories = items
                .GetFacet("categories")
                .ToDictionary(x => UmbracoContext.Content.GetById(Udi.Parse(x.Value.ToString())), x => x.Hits);
```